src/System.Runtime.Extensions/tests/System/Environment.MachineName.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace System.Tests
+{
+    public class Environment_MachineName
+    {
+        // [Fact] Skip test for the mscolib changes to reach corefx
+        public void TestMachineNameProperty()
+        {
+            string computerName = GetComputerName();
+            Assert.Equal(computerName, Environment.MachineName);
+        }
+
+        [DllImport("kernel32.dll", EntryPoint = "GetComputerNameW")]
+        private unsafe static extern int GetComputerName(char* nameBuffer, ref int bufferSize);
+
+        [DllImport("libc")]
+        private unsafe static extern int gethostname(char* name, int len);
+
+        private unsafe static string GetComputerName()
+        {
+            const int MaxMachineNameLength = 256;
```

Comment:
It is pointless to test the API by duplicating its exact implementation in the test.

A better test for this API would:
- On Windows, compare the returned value with COMPUTERNAME environment variable
- On Unix, compare the returned value with HOSTNAME environment variable